src/os/layer/preset/presetmenubutton.js

```diff
@@ -258,14 +346,17 @@ class Controller extends MenuButtonCtrl {
     const source = this.scope['preset'];
     const preset = this.clone(source);
 
-    // get PresetService(s) that support save()
-    const services = LayerPresetManager.getInstance().supporting(OsLayerPreset.PresetServiceAction.UPDATE);
-    if (services && services.length > 0) {
-      // TODO if more than one, open a dropdown modal to select one
-      services[0].update(preset).then(this.saveSuccess.bind(this), this.saveFailure.bind(this));
-    } else {
-      GoogLog.error(LOGGER, 'No services found that support SAVE action');
-    }
+    // get PresetService(s) that support uodate()
+    this.getService(OsLayerPreset.PresetServiceAction.UPDATE).then(
+        (service) => {
+          service.update(preset).then(
+              this.saveSuccess.bind(this),
+              this.onServiceFailure.bind(this, 'Could not save Preset'));
+        },
+        (msg) => {
+          GoogLog.error(LOGGER, 'No services found that support save action');
```

Comment:
This should include the `msg` argument value in its logged message.

Reply:
Updated. Ironically, msg was basically the same thing, so I just used it instead, lol.